sdk/tests/js/tests/tex-image-and-sub-image-utils.js

```diff
@@ -791,11 +791,71 @@ var TexImageUtils = (function() {
     return program;
   };
 
+  /**
+   * Return a list of unpack color spaces to test, supported by the specified
+   * WebGLRenderingContext.
+   */
+  var unpackColorSpacesToTest = function(gl)
+  {
+    if ('unpackColorSpace' in gl)
+      return ['srgb', 'display-p3'];
+    else
+      return [null];
+  }
+
+  /**
+   * For each entry in unpackColorSpaces, duplicate all of cases, adding an
+   * unpackColorSpace key with its value set to that entry to each case.
+   */
+  var crossProductTestCasesWithUnpackColorSpaces = function(cases, unpackColorSpaces)
+  {
+    var caseWithColorSpace = function(c, cs) { return Object.assign({}, c, {unpackColorSpace:cs}); }
```

Comment:
```
return Object.assign({}, c, {unpackColorSpace:cs})
```

can be shortened to

```
return {...c, ...{unpackColorSpace:cs}};
```

Reply:
This didn't work when I tested against FF, so I left it unchanged.

Reply:
How did you test? Works in Firefox just fine AFAIK

https://jsgist.org/?src=c212a3fa63ff82bc55c88b73fd996b1b

I've been using it for years

Reply:
Hmm, no idea what I did wrong. Now it works fine. Updated.